vcd/resource_vcd_rde_type.go

```diff
@@ -0,0 +1,344 @@
+package vcd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+func resourceVcdRdeType() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdRdeTypeCreate,
+		ReadContext:   resourceVcdRdeTypeRead,
+		UpdateContext: resourceVcdRdeTypeUpdate,
+		DeleteContext: resourceVcdRdeTypeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdRdeTypeImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the Runtime Defined Entity Type",
+			},
+			"vendor": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`(?i)^[a-z0-9_-]+$`), "only alphanumeric characters, underscores and hyphens allowed"),
```

Comment:
Since this validation is used several times, we should make a permanent one in `validate_func.go`

Reply:
True, I refactored it into the `validateAlphanumericWithUnderscoresAndHyphens` function.